bloom_nofos/users/views.py

```diff
@@ -66,3 +71,103 @@ def post(self, request):
 
         return render(request, self.template_name, {"form": form})
+
+
+def login_view(request):
+    """Initiate Login.gov authentication flow."""
+    client = LoginGovClient()
+    auth_url, state, nonce = client.get_authorization_url()
+
+    # Store state and nonce in session for validation in callback
+    request.session["login_gov_state"] = state
+    request.session["login_gov_nonce"] = nonce
+
+    return redirect(auth_url)
+
+
+@require_http_methods(["GET"])
+def callback(request):
+    """Handle Login.gov callback."""
+    error = request.GET.get("error")
+    if error:
+        messages.error(request, f"Login.gov error: {error}")
+        return redirect(settings.LOGIN_URL)
+
+    # Validate state
+    state = request.GET.get("state")
+    stored_state = request.session.get("login_gov_state")
+    if not state or state != stored_state:
+        messages.error(request, "Invalid state parameter")
+        return redirect(settings.LOGIN_URL)
+
+    # Get the authorization code
+    code = request.GET.get("code")
+    if not code:
+        messages.error(request, "No authorization code received")
+        return redirect(settings.LOGIN_URL)
+
+    try:
+        # Exchange code for tokens
+        client = LoginGovClient()
+        token_response = client.get_token(code)
+
+        # Validate ID token
+        id_token = token_response.get("id_token")
+        if not id_token:
+            messages.error(request, "No ID token in response")
+            return redirect(settings.LOGIN_URL)
+
+        stored_nonce = request.session.get("login_gov_nonce")
+        user_data = client.validate_id_token(id_token, stored_nonce)
+
+        # Authenticate user
+        user = authenticate(request, login_gov_data=user_data)
+        if user:
+            login(request, user)
+
+            # Clean up session
+            request.session.pop("login_gov_state", None)
+            request.session.pop("login_gov_nonce", None)
+
+            # Redirect to next URL if available, otherwise to default
+            next_url = request.session.get("next", settings.LOGIN_REDIRECT_URL)
+            return redirect(next_url)
+        else:
+            messages.error(request, "Authentication failed")
+
+    except Exception as e:
+        if "Private key not configured" in str(e):
+            messages.error(
+                request,
+                "Login.gov integration is not fully configured. Please contact your administrator.",
+            )
+        else:
+            messages.error(request, f"Login failed: {str(e)}")
+
+    return redirect(settings.LOGIN_URL)
+
+
+def logout_view(request):
+    """Handle user logout."""
+    logout(request)
+    return redirect(settings.LOGOUT_REDIRECT_URL)
+
+
+def traditional_login_view(request):
+    """Handle traditional login with username/password."""
+    form = LoginForm(request.POST or None)
+
+    if request.method == "POST":
+        if form.is_valid():
+            email = form.cleaned_data["email"]
```

Comment:
Let's make sure we lowercase the emails people use: `email = form.cleaned_data["email"].strip().lower()`

Some HRSA emails are sent to me like `PCraig@hrsa.gov`, but I always enter them as lowercase.